From dd4c40fae63a5fcfe5b10623f8d6a72a985b321a Mon Sep 17 00:00:00 2001
From: Arik Nemtsov <arik@wizery.com>
Date: Wed, 25 Jul 2012 17:56:08 +0300
Subject: [PATCH] WifiP2PService: unset the interface address when stopping GO

This way if a P2P CL is started on the same interface the GO previously
was, we will be able to get a DHCP address.

Change-Id: I81d302fc845e58939991a4adf8c11c2c9562c02e
Signed-off-by: Arik Nemtsov <arik@wizery.com>
---
 wifi/java/android/net/wifi/p2p/WifiP2pService.java |    1 +
 1 file changed, 1 insertion(+)

diff --git a/wifi/java/android/net/wifi/p2p/WifiP2pService.java b/wifi/java/android/net/wifi/p2p/WifiP2pService.java
index 9e004d0..b29feb4 100644
--- a/wifi/java/android/net/wifi/p2p/WifiP2pService.java
+++ b/wifi/java/android/net/wifi/p2p/WifiP2pService.java
@@ -1414,6 +1414,7 @@ public class WifiP2pService extends IWifiP2pManager.Stub {
 
     private void stopDhcpServer() {
         try {
+            mNwService.clearInterfaceAddresses(mInterface);
             mNwService.stopTethering();
         } catch (Exception e) {
             loge("Error stopping Dhcp server" + e);
-- 
1.7.9.5

